src/lib/y2network/interfaces.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "forwardable"
+
+module Y2Network
+  # A container for network devices. In the end should carry methods for mass operations
+  # over network interfaces like old LanItems::find_dhcp_ifaces or so.
+  #
+  # FIXME: Intended for LanItems::Items separation
+  # proper cleanup is must
+  class Interfaces
```

Comment:
I like the idea of having a container for interfaces (as we have in [storage-ng](https://github.com/yast/yast-storage-ng/blob/5a8963f0c4b5a242ac8d3122ca5225e7f216e6a4/src/lib/y2storage/planned/devices_collection.rb)), although I would call it `InterfacesCollection` or `InterfacesList`.

However, this class is responsible for converting the information from sysconfig into `Interface` objects too (it receives the result of calling [Yast::LanItems.Items](https://github.com/yast/yast-network/pull/796/files#diff-66d00977c16e82a3b5f1046c2e5feb60R39)) which looks wrong to me. What if tomorrow we want to support any other technology? I think it should receive an array of `Interface` objects.

Then, who is responsible for building the `Interface` objects? I think `ConfigReader::Sysconfig` (or any ancillary class) is.

Reply:
we can separate this conversion method to another kind of reader for example. I'm not against that.

Renaming from ```Interfaces``` to e.g. ```InterfacesCollection``` also makes sense.